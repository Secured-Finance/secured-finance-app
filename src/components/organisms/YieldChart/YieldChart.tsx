import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import SFLogoSmall from 'src/assets/img/logo-small.svg';
import { ExpandIndicator } from 'src/components/atoms';
import { CurveHeader, LineChart, getData } from 'src/components/molecules';
import {
    selectLandingOrderForm,
    setMaturity,
} from 'src/store/landingOrderForm';
import { RootState } from 'src/store/types';
import { DailyVolumes, MaturityOptionList } from 'src/types';
import { CurrencySymbol, Rate } from 'src/utils';
import { Maturity } from 'src/utils/entities';

interface YieldChartProps {
    asset: CurrencySymbol;
    isBorrow: boolean;
    rates: Array<Rate>;
    maturitiesOptionList: MaturityOptionList;
    dailyVolumes: DailyVolumes;
}

export const YieldChart = ({
    asset,
    isBorrow,
    rates,
    maturitiesOptionList,
    dailyVolumes,
}: YieldChartProps): JSX.Element => {
    const dispatch = useDispatch();

    const { maturity } = useSelector((state: RootState) =>
        selectLandingOrderForm(state.landingOrderForm)
    );
    const [show, setShow] = useState(true);

    return (
        <div className='flex h-[480px] w-fit flex-row overflow-hidden rounded-r-xl border-y border-r border-panelStroke shadow-[0_46px_64px_rgba(0,0,0,0.4)]'>
            <div
                className={`flex flex-col items-start overflow-hidden transition-width duration-700 ease-out ${
                    show ? 'w-[640px] pl-8' : 'w-0'
                }`}
                data-testid='yield-chart-component'
            >
                <div className='h-20'>
                    <CurveHeader
                        asset={asset}
                        dailyVolumes={dailyVolumes}
                    ></CurveHeader>
                </div>
                <div className='flex flex-grow items-center pl-[35px]'>
                    <div className='h-[350px] w-[500px]'>
                        {rates && (
                            <LineChart
                                type='line'
                                data={getData(
                                    rates,
                                    isBorrow ? 'Borrow' : 'Lend',
                                    maturitiesOptionList.map(o => o.label)
                                )}
                                maturitiesOptionList={maturitiesOptionList}
                                handleChartClick={(maturity: Maturity) =>
                                    dispatch(setMaturity(maturity))
                                }
                                maturity={
                                    maturity.isZero()
                                        ? maturitiesOptionList[0].value
                                        : maturity
                                }
                            />
                        )}
                    </div>
                </div>
            </div>
            <button
                className='relative flex w-10 items-center justify-center bg-starBlue-30'
                onClick={() => setShow(!show)}
            >
                <SFLogoSmall className='absolute top-2 h-8 w-8' />
                <div className='rotate-90'>
                    <ExpandIndicator expanded={!show} />
                </div>
            </button>
        </div>
    );
};
