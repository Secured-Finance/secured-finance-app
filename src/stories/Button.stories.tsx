import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Button, IButton } from 'src/components/new/Button';

export default {
    title: 'Components/Button',
    component: Button,
    argTypes: { onClick: { action: 'clicked' } },
} as Meta;

const Template: Story<IButton> = args => <Button {...args}>Button</Button>;

export const Default = Template.bind({});
Default.args = {
    outline: false,
};

export const Outlined = Template.bind({});
Outlined.args = {
    outline: true,
};
