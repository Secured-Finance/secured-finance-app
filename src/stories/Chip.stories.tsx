import React from 'react';
import { Story, Meta } from '@storybook/react';
import cm from './index.module.scss';

import { ChipButton, IChipButton } from 'src/components/new/Chip/ChipButton';
import { ChipLabel, IChipLabel } from 'src/components/new/Chip/ChipLabel';

export default {
    title: 'Components/Chips',
    argTypes: { onClick: { action: 'clicked' } },
} as Meta;

export const ChipButtons: Story<IChipButton> = () => (
    <div className={cm.elementsContainer}>
        <ChipButton>Button</ChipButton>
        <ChipButton accent={'purple'}>Button</ChipButton>
        <ChipButton accent={'red'}>Button</ChipButton>
    </div>
);

export const ChipLabels: Story<IChipLabel> = () => (
    <div className={cm.elementsContainer}>
        <ChipLabel label={'Registered'} />
        <ChipLabel label={'Working'} accent={'green'} />
        <ChipLabel label={'Terminated'} accent={'red'} />
        <ChipLabel label={'Closed'} accent={'grey'} />
        <ChipLabel label={'Past Due'} accent={'turquoise'} />
        <ChipLabel label={'Due'} accent={'blue'} />
    </div>
);
