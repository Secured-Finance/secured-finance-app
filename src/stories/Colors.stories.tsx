import { Meta, Story } from '@storybook/react';

export default {
    title: 'Design System/Colors',
    parameters: {
        chromatic: { disableSnapshot: false },
    },
} as Meta;

export const Primary: Story = () => (
    <div className='inline-flex flex-col items-center justify-start rounded-3xl bg-white p-10'>
        <div className='flex flex-col items-start justify-start space-y-6'>
            <p className='text-lg font-semibold leading-10 text-black-80'>
                Primary
            </p>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg bg-universeBlue' />
                <div className='h-10 w-10 rounded-lg bg-horizonBlue' />
                <div className='h-10 w-10 rounded-lg bg-starBlue' />
                <div className='h-10 w-10 rounded-lg bg-teal' />
                <div className='h-10 w-10 rounded-lg bg-green' />
                <div className='h-10 w-10 rounded-lg bg-orange' />
                <div className='h-10 w-10 rounded-lg bg-red' />
            </div>
            <div className='h-10 w-10 rounded-lg bg-purple' />
        </div>
    </div>
);

export const Secondary: Story = () => (
    <div className='inline-flex flex-col items-start justify-start space-y-6 rounded-3xl bg-white p-10'>
        <p className='text-lg font-semibold leading-10 text-black-80'>
            Secondary
        </p>
        <div className='inline-flex items-center justify-start space-x-6'>
            <div className='h-10 w-10 rounded-lg bg-galacticOrange' />
            <div className='h-10 w-10 rounded-lg bg-planetaryPurple' />
            <div className='h-10 w-10 rounded-lg bg-nebulaTeal' />
            <div className='h-10 w-10 rounded-lg bg-secondaryPurple ' />
            <div className='h-10 w-10 rounded-lg bg-moonGrey' />
        </div>
    </div>
);

export const Shades: Story = () => (
    <div className='inline-flex flex-col items-center justify-start rounded-3xl bg-white p-10'>
        <div className='flex flex-col items-start justify-start space-y-6'>
            <p className='text-lg font-semibold leading-10 text-black-80'>
                Shades
            </p>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg bg-starBlue-80' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-60' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-40' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-20' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-10' />
            </div>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg bg-black-90' />
                <div className='h-10 w-10 rounded-lg bg-black-80' />
                <div className='h-10 w-10 rounded-lg bg-black-70' />
                <div className='h-10 w-10 rounded-lg bg-black-60' />
                <div className='h-10 w-10 rounded-lg bg-black-50' />
                <div className='h-10 w-10 rounded-lg bg-black-40' />
                <div className='h-10 w-10 rounded-lg bg-black-30' />
                <div className='h-10 w-10 rounded-lg bg-black-20' />
                <div className='h-10 w-10 rounded-lg bg-black-10' />
            </div>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg border-2 bg-white-90' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-80' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-70' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-60' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-50' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-40' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-30' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-20' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-10' />
            </div>
        </div>
    </div>
);
