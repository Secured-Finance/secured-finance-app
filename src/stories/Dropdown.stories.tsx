import React from 'react';
import { Story, Meta } from '@storybook/react';
import { Dropdown, IDropdown } from 'src/components/new/Dropdown';
import { filecoin, telegram } from 'src/components/new/icons';

export default {
    title: 'Components/Dropdown',
    component: Dropdown,
    argTypes: { onClick: { action: 'clicked' } },
} as Meta;

const Template: Story<IDropdown> = args => {
    const [value, setValue] = React.useState(args.value ?? '');

    return (
        <Dropdown
            {...args}
            onChange={e => {
                setValue(e.target.value);
            }}
            value={value}
        />
    );
};

let value = 1;
export const Default = Template.bind({});
Default.args = {
    options: [
        { value: 1, label: 'One' },
        { value: 2, label: 'Two', icon: filecoin },
        { value: 3, label: 'Three looooong' },
        { value: 4, label: 'Long with icon', icon: telegram },
    ],
    value,
    label: '',
    noBorders: false,
};
