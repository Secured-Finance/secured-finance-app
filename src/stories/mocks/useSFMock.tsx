import { Currency } from '@secured-finance/sf-core';
import { BigNumber } from 'ethers';
import * as jest from 'jest-mock';
import { CurrencySymbol, getCurrencyMapAsList } from 'src/utils';
import { collateralBook80, dec24Fixture } from './fixtures';
import {
    dec22Fixture,
    mar23Fixture,
    ethBytes32,
    efilBytes32,
} from './fixtures';

export const mockUseSF = () => {
    const mockSecuredFinance = {
        placeOrder: jest.fn(),
        placePreOrder: jest.fn(),
        getBorrowUnitPrices: jest.fn(() =>
            Promise.resolve([
                BigNumber.from(9687),
                BigNumber.from(9685),
                BigNumber.from(9679),
                BigNumber.from(9674),
                BigNumber.from(9653),
                BigNumber.from(9643),
                BigNumber.from(9627),
                BigNumber.from(9617),
            ])
        ),
        getLendUnitPrices: jest.fn(() =>
            Promise.resolve([
                BigNumber.from(9685),
                BigNumber.from(9683),
                BigNumber.from(9677),
                BigNumber.from(9672),
                BigNumber.from(9651),
                BigNumber.from(9641),
                BigNumber.from(9625),
                BigNumber.from(9615),
            ])
        ),
        getMidUnitPrices: jest
            .fn<Promise<BigNumber[]>, Currency[]>()
            .mockImplementation((ccy: Currency) => {
                switch (ccy.symbol) {
                    case CurrencySymbol.ETH:
                        return Promise.resolve([
                            BigNumber.from(9686),
                            BigNumber.from(9684),
                            BigNumber.from(9678),
                            BigNumber.from(9673),
                            BigNumber.from(9652),
                            BigNumber.from(9642),
                            BigNumber.from(9626),
                            BigNumber.from(9616),
                        ]);
                    case CurrencySymbol.EFIL:
                        return Promise.resolve([
                            BigNumber.from(9586),
                            BigNumber.from(9584),
                            BigNumber.from(9578),
                            BigNumber.from(9573),
                            BigNumber.from(9552),
                            BigNumber.from(9542),
                            BigNumber.from(9526),
                            BigNumber.from(9516),
                        ]);
                    case CurrencySymbol.USDC:
                        return Promise.resolve([
                            BigNumber.from(9486),
                            BigNumber.from(9484),
                            BigNumber.from(9478),
                            BigNumber.from(9473),
                            BigNumber.from(9452),
                            BigNumber.from(9442),
                            BigNumber.from(9426),
                            BigNumber.from(9416),
                        ]);
                    case CurrencySymbol.WBTC:
                        return Promise.resolve([
                            BigNumber.from(9386),
                            BigNumber.from(9384),
                            BigNumber.from(9378),
                            BigNumber.from(9373),
                            BigNumber.from(9352),
                            BigNumber.from(9342),
                            BigNumber.from(9326),
                            BigNumber.from(9316),
                        ]);
                    default:
                        throw new Error('Not implemented');
                }
            }),
        getCollateralBook: jest.fn(() =>
            Promise.resolve({
                collateral: {
                    ...collateralBook80.collateral,
                    ...collateralBook80.nonCollateral,
                },
                collateralCoverage: collateralBook80.coverage,
            })
        ),

        getLendingMarket: jest.fn((_, maturity: number) => {
            if (maturity === dec24Fixture.toNumber()) {
                return Promise.resolve({
                    isItayosePeriod: jest.fn(() => Promise.resolve(false)),
                    isOpened: jest.fn(() => Promise.resolve(false)),
                    isPreOrderPeriod: jest.fn(() => Promise.resolve(true)),
                });
            } else {
                return Promise.resolve({
                    isItayosePeriod: jest.fn(() => Promise.resolve(false)),
                    isOpened: jest.fn(() => Promise.resolve(true)),
                    isPreOrderPeriod: jest.fn(() => Promise.resolve(false)),
                });
            }
        }),

        getMaturities: jest.fn(() =>
            Promise.resolve([
                BigNumber.from('1000'),
                BigNumber.from('2000'),
                BigNumber.from('3000'),
                BigNumber.from('4000'),
                BigNumber.from('5000'),
            ])
        ),

        getLendingMarkets: jest.fn(() =>
            Promise.resolve([
                {
                    midRate: 100,
                    lendRate: 200,
                    borrowRate: 300,
                    maturity: 1000,
                    name: 'ETH-1000',
                    utcOpeningDate: 1620000000,
                },
                {
                    midRate: 100,
                    lendRate: 200,
                    borrowRate: 300,
                    maturity: 2000,
                    name: 'ETH-2000',
                    utcOpeningDate: 1720000000,
                },
            ])
        ),

        depositCollateral: jest.fn(() =>
            Promise.resolve({
                wait: jest.fn(() => Promise.resolve({ blockNumber: 123 })),
                to: '0xb98bd7c7f656290071e52d1aa617d9cb4467fd6d',
            })
        ),

        withdrawCollateral: jest.fn<
            Promise<{
                wait: () => Promise<{
                    blockNumber: number | undefined;
                }>;
            }>,
            []
        >(() =>
            Promise.resolve({
                wait: () => Promise.resolve({ blockNumber: 123 }),
            })
        ),

        getBorrowOrderBook: jest.fn(() =>
            Promise.resolve({
                unitPrices: [
                    BigNumber.from(9690),
                    BigNumber.from(9687),
                    BigNumber.from(9685),
                    BigNumber.from(9679),
                    BigNumber.from(9674),
                ],
                amounts: [
                    BigNumber.from('43000000000000000000000'),
                    BigNumber.from('23000000000000000000000'),
                    BigNumber.from('15000000000000000000000'),
                    BigNumber.from('12000000000000000000000'),
                    BigNumber.from('1800000000000000000000'),
                ],
                quantities: [
                    BigNumber.from('1000'),
                    BigNumber.from('2000'),
                    BigNumber.from('3000'),
                    BigNumber.from('4000'),
                    BigNumber.from('5000'),
                ],
            })
        ),

        getLendOrderBook: jest.fn(() =>
            Promise.resolve({
                unitPrices: [
                    BigNumber.from(9690),
                    BigNumber.from(9687),
                    BigNumber.from(9685),
                    BigNumber.from(9679),
                    BigNumber.from(9674),
                ],
                amounts: [
                    BigNumber.from('43000000000000000000000'),
                    BigNumber.from('23000000000000000000000'),
                    BigNumber.from('15000000000000000000000'),
                    BigNumber.from('12000000000000000000000'),
                    BigNumber.from('1800000000000000000000'),
                ],
                quantities: [
                    BigNumber.from('1000'),
                    BigNumber.from('2000'),
                    BigNumber.from('3000'),
                    BigNumber.from('4000'),
                    BigNumber.from('5000'),
                ],
            })
        ),

        getERC20Balance: jest.fn(() => Promise.resolve({ balance: 10000000 })),

        cancelLendingOrder: jest.fn(() =>
            Promise.resolve({
                wait: jest.fn(() => Promise.resolve({ blockNumber: 123 })),
            })
        ),

        getERC20TokenContractAddress: jest.fn(() =>
            Promise.resolve('0xEd4733fE7BAc4C2934F7e9CE4e0696b2169701D8')
        ),

        mintERC20Token: jest.fn(() =>
            Promise.resolve({
                wait: jest.fn(() => Promise.resolve({ blockNumber: 123 })),
            })
        ),

        getCurrencies: jest.fn(() =>
            Promise.resolve(
                getCurrencyMapAsList().map(currency => currency.symbol)
            )
        ),

        getProtocolDepositAmount: jest.fn(() =>
            Promise.resolve({
                ETH: BigNumber.from('100000000000000000000'), // 100 ETH
                EFIL: BigNumber.from('100000000000000000000000'), // 100 000 EFIL
                USDC: BigNumber.from('1000000000000'), // 1 000 000 USDC
                WBTC: BigNumber.from('1000000000000'), // 1000 BTC
            })
        ),

        unwindOrder: jest.fn(() =>
            Promise.resolve({
                hash: '0x123',
                wait: jest.fn(() =>
                    Promise.resolve({
                        blockNumber: 123,
                    })
                ),
            })
        ),

        getCollateralParameters: jest.fn(() =>
            Promise.resolve({
                liquidationThresholdRate: BigNumber.from('12500'),
            })
        ),

        getUsedCurrenciesForOrders: jest.fn(() =>
            Promise.resolve([ethBytes32, efilBytes32])
        ),

        getOrderList: jest.fn(() =>
            Promise.resolve({
                activeOrders: [
                    {
                        orderId: 1,
                        ccy: ethBytes32,
                        side: 0,
                        maturity: BigNumber.from(dec22Fixture.toString()),
                        unitPrice: BigNumber.from('9800'),
                        amount: BigNumber.from('1000000000'),
                        timestamp: BigNumber.from('1609295092'),
                    },
                    {
                        orderId: 2,
                        ccy: ethBytes32,
                        side: 0,
                        maturity: BigNumber.from(mar23Fixture.toString()),
                        unitPrice: BigNumber.from('9800'),
                        amount: BigNumber.from('1000000000'),
                        timestamp: BigNumber.from('1609295092'),
                    },
                    {
                        orderId: 1,
                        ccy: efilBytes32,
                        side: 1,
                        maturity: BigNumber.from(dec22Fixture.toString()),
                        unitPrice: BigNumber.from('9800'),
                        amount: BigNumber.from('1000000000'),
                        timestamp: BigNumber.from('1609295092'),
                    },
                ],
                inactiveOrders: [
                    {
                        orderId: 4,
                        ccy: ethBytes32,
                        side: 0,
                        maturity: BigNumber.from(dec22Fixture.toString()),
                        unitPrice: BigNumber.from('9800'),
                        amount: BigNumber.from('1000000000'),
                        timestamp: BigNumber.from('1609295092'),
                    },
                    {
                        orderId: 5,
                        ccy: efilBytes32,
                        side: 0,
                        maturity: BigNumber.from(dec22Fixture.toString()),
                        unitPrice: BigNumber.from('9800'),
                        amount: BigNumber.from('1000000000'),
                        timestamp: BigNumber.from('1609295092'),
                    },
                ],
            })
        ),
    };

    return mockSecuredFinance;
};
