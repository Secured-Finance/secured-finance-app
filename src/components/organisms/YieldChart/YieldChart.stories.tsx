import { ComponentMeta, ComponentStory } from '@storybook/react';
import { withAssetPrice } from 'src/../.storybook/decorators';
import { dailyVolumes, maturityOptions } from 'src/stories/mocks/fixtures';
import { CurrencySymbol, Rate } from 'src/utils';
import { YieldChart } from './';

const rates = [
    new Rate(10000),
    new Rate(20000),
    new Rate(30000),
    new Rate(40000),
    new Rate(50000),
    new Rate(60000),
    new Rate(70000),
    new Rate(80000),
];
export default {
    title: 'Organism/YieldChart',
    component: YieldChart,
    chromatic: { pauseAnimationAtEnd: true, diffThreshold: 1 },
    args: {
        asset: CurrencySymbol.EFIL,
        isBorrow: true,
        rates: rates,
        maturitiesOptionList: maturityOptions,
        dailyVolumes: dailyVolumes,
    },
    argTypes: {},
    decorators: [withAssetPrice],
} as ComponentMeta<typeof YieldChart>;

const Template: ComponentStory<typeof YieldChart> = args => {
    return <YieldChart {...args} />;
};

export const Default = Template.bind({});
export const Loading = Template.bind({});
Loading.args = {
    rates: [],
};
