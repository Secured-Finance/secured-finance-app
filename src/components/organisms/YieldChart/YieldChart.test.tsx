import { composeStories } from '@storybook/react';
import { preloadedAssetPrices } from 'src/stories/mocks/fixtures';
import { mockUseSF } from 'src/stories/mocks/useSFMock';
import {
    fireEvent,
    queries,
    render,
    RenderResult,
    screen,
    waitFor,
} from 'src/test-utils.js';
import * as stories from './YieldChart.stories';

const { Default, Loading } = composeStories(stories);

const mockSecuredFinance = mockUseSF();
jest.mock('src/hooks/useSecuredFinance', () => () => mockSecuredFinance);

describe('YieldChart Component', () => {
    it('should render YieldChart', async () => {
        let ag: RenderResult<typeof queries, HTMLElement>;
        await waitFor(() => {
            ag = render(<Default />, {
                preloadedState: {
                    ...preloadedAssetPrices,
                },
            });
        }).then(() => {
            expect(ag.baseElement).toMatchSnapshot();
        });
    });

    it('should close and open YieldChart on button click', async () => {
        render(<Default />);
        const btn = screen.getByRole('button');
        expect(screen.getByTestId('yield-chart-component')).toHaveClass(
            'w-[640px]'
        );

        fireEvent.click(btn);
        expect(screen.getByTestId('yield-chart-component')).toHaveClass('w-0');

        fireEvent.click(btn);
        expect(screen.getByTestId('yield-chart-component')).toHaveClass(
            'w-[640px]'
        );
    });

    it('should show the spinner when loading', async () => {
        render(<Loading />);
        expect(
            screen.getByRole('alertdialog', { name: 'Loading' })
        ).toBeInTheDocument();
    });
});
