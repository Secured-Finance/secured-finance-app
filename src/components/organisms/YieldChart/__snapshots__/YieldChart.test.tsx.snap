// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YieldChart Component should render YieldChart 1`] = `
<body>
  <div>
    <div
      class="flex h-[480px] w-fit flex-row overflow-hidden rounded-r-xl border-y border-r border-panelStroke shadow-[0_46px_64px_rgba(0,0,0,0.4)]"
    >
      <div
        class="flex flex-col items-start overflow-hidden transition-width duration-700 ease-out w-[640px] pl-8"
        data-testid="yield-chart-component"
      >
        <div
          class="h-20"
        >
          <div
            class="flex h-20 w-[585px] flex-row justify-between p-4"
          >
            <div
              class="flex h-[44px] w-fit flex-row items-center justify-between gap-3"
            >
              <div>
                <svg
                  aria-label="EFIL"
                  class="h-9 w-9"
                  role="img"
                />
              </div>
              <div
                class="flex h-full w-full flex-col"
              >
                <span
                  class="typography-caption-2 h-5 text-planetaryPurple"
                >
                  Filecoin
                </span>
                <div
                  class="flex h-6 w-full flex-row items-center justify-between gap-3"
                >
                  <span
                    class="typography-caption flex h-full items-center text-neutral-8"
                  >
                    $6.00
                  </span>
                  <span
                    class="typography-caption flex h-full items-center text-galacticOrange"
                  >
                    -8.21%
                  </span>
                </div>
              </div>
            </div>
            <div
              class="flex flex-row gap-2"
            >
              <div
                class="flex h-[44px] w-[135px] flex-col p-0 text-center"
              >
                <div
                  class="typography-caption-2 h-5 text-planetaryPurple"
                >
                  Total Volume (Asset)
                </div>
                <div
                  class="typography-caption h-6 text-white-60"
                >
                  657,000 EFIL
                </div>
              </div>
              <div
                class="flex h-[44px] w-[135px] flex-col p-0 text-center"
              >
                <div
                  class="typography-caption-2 h-5 text-planetaryPurple"
                >
                  Total Volume (USD)
                </div>
                <div
                  class="typography-caption h-6 text-white-60"
                >
                  $3,942,000
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-grow items-center pl-[35px]"
        >
          <div
            class="h-[350px] w-[500px]"
          >
            <canvas
              data-chromatic="ignore"
              height="0"
              role="img"
              style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
              width="0"
            />
          </div>
        </div>
      </div>
      <button
        class="relative flex w-10 items-center justify-center bg-starBlue-30"
      >
        <svg
          class="absolute top-2 h-8 w-8"
        />
        <div
          class="rotate-90"
        >
          <svg
            aria-hidden="true"
            class="h-5 w-5 text-white"
            data-testid="chevron-down-icon"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 9l-7 7-7-7"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</body>
`;
